File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] a) {
        int ans = a[0];
        int cr = ans ;
        for(int i = 1 ;i < a.length ; i ++){
            if(a[i] > a[i - 1]){
                cr += a[i];
            }else{
                cr = a[i];
            }
            ans = Math.max(ans, cr);
        }
        return ans ;
    }
}

File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, a: List[int]) -> int:
        ans = cr = a[0]
        for i in range(1, len(a)):
            if a[i]  > a[i - 1]:
                cr += a[i]
            else: cr = a[i]
            ans = max(ans, cr)
        return ans 